### PR TITLE
[GOBBLIN-1080]Add configuration to preserve schema creation time in converter

### PR DESCRIPTION
**Pull request by ZihanLi58**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1080

### Description
- [ ] Here are some details about my PR, including screenshots (if applicable):
Add configuration to preserve schema creation time in converter, which can be used as schema version

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Unit test

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-api/src/main/java/org/apache/gobblin/configuration/ConfigurationKeys.java [hunk 1]
```diff
@@ -308,6 +308,8 @@
   /**
    * Converter configuration properties.
    */
+  public static final String CONVERTER_PRESERVE_SCHEMA_CREATION_TIME = "converter.preserve.schema.creation.time";
```
Comment (sv2000) [inline gobblin-api/src/main/java/org/apache/gobblin/configuration/ConfigurationKeys.java]: "converter.avro.include.schema.creation.time"?

File: gobblin-utility/src/main/java/org/apache/gobblin/util/AvroUtils.java [hunk 2]
```diff
@@ -116,6 +117,13 @@ public static boolean checkReaderWriterCompatibility(Schema readerSchema, Schema
     return SchemaCompatibility.checkReaderWriterCompatibility(readerSchema, writerSchema).getType().equals(SchemaCompatibility.SchemaCompatibilityType.COMPATIBLE);
   }
 
+  public static Schema preserveCreationTime(Schema inputSchema, Schema outputSchema) {
```
Comment (sv2000) [inline gobblin-utility/src/main/java/org/apache/gobblin/util/AvroUtils.java]: preserveCreationTime -> addSchemaCreationTime?

**Comment by sv2000** [comment]
@ZihanLi58 check on the travis failure.

**Comment by autumnust** [comment]
Can you rebase your changes against master ? Looks like it pulls in many other commits

**Pull request closed without merging**